### PR TITLE
TreeNode, Collapsing Headers, Selectable and Flags

### PR DESCRIPTION
**Pull request by supudo**
Hi :)

I've added some more widgets and "templated" the docs for the new flags added.
You might want to check the implementation of the listbox() function - my skills in Cython are not that good, so i might've gotten wrong the conversion from Python list to const char**.

Again - if you have comments i'll be glad to work on them.

S.

### CODE REVIEW

File: imgui/core.pyx [hunk 1]
```diff
@@ -1374,6 +1393,307 @@ def is_window_collapsed():
...
+        &inout_selected,
+        flags,
+        _cast_args_ImVec2(width, height)), inout_selected
+
+
+def listbox(
+    str label,
+    int current,
+    list items,
+    int items_count,
```
Comment (swistakm) [inline imgui/core.pyx]: Do we need this argument? We could simply use `len(items)` instead. If user wants to limit number of displayed items he easily slice list.

File: imgui/core.pyx [hunk 2]
```diff
@@ -1374,6 +1393,307 @@ def is_window_collapsed():
...
+    int height_in_items=-1
+):
+    """Show listbox widget.
+
+    .. visual-example::
+        :auto_layout:
+        :height: 100
```
Comment (swistakm) [inline imgui/core.pyx]: add `:width: 200` for better display.

File: imgui/core.pyx [hunk 3]
```diff
@@ -1374,6 +1393,307 @@ def is_window_collapsed():
...
+    cdef const char** in_items = <const char**> malloc(len(items) * sizeof(char*))
+    for i in range(len(items)):
+        in_items[i] = strdup(_bytes(items[i]))
+
+    opened = cimgui.ListBox(
+        _bytes(label),
+        &inout_current,
+        in_items,
+        items_count,
+        height_in_items)
```
Comment (swistakm) [inline imgui/core.pyx]: move trailing parens to next line

File: imgui/core.pyx [hunk 4]
```diff
@@ -1374,6 +1393,307 @@ def is_window_collapsed():
...
+            int* current_item,
+            const char** items,
+            int items_count,
+            int height_in_items = -1
+
+    """
+    cdef int inout_current = current
+
+    cdef const char** in_items = <const char**> malloc(len(items) * sizeof(char*))
+    for i in range(len(items)):
```
Comment (swistakm) [inline imgui/core.pyx]: Can't we use `enumerate`?

```python
for index, item in enumerate(items):
    in_items[index] = strdup(_bytes(item))
```

File: imgui/core.pyx [hunk 5]
```diff
@@ -1374,6 +1393,307 @@ def is_window_collapsed():
...
+        bool ListBox(
+            const char* label,
+            int* current_item,
+            const char** items,
+            int items_count,
+            int height_in_items = -1
+
+    """
+    cdef int inout_current = current
+
```
Comment (swistakm) [inline imgui/core.pyx]: keep `cdef`s together: remove this empty line but add new after last `cdef`

File: imgui/core.pyx [hunk 6]
```diff
@@ -1374,6 +1393,307 @@ def is_window_collapsed():
...
+        in_items[i] = strdup(_bytes(items[i]))
+
+    opened = cimgui.ListBox(
+        _bytes(label),
+        &inout_current,
+        in_items,
+        items_count,
+        height_in_items)
+
+    free(in_items)
```
Comment (swistakm) [inline imgui/core.pyx]: There is a memory issue. You have freed `in_items` array but have not freed each string created by `strdup` separately.

**Comment by swistakm** [comment]
Good work. Mostly documentation issues. There is only one issue that I am really concerned about: the memory leak in `listbox()` function.

**Comment by supudo** [comment]
Think i nailed them all. Committed to PR#15.

S.

**Comment by supudo** [comment]
A quick question here while we're on the memory topic.

The combo method is calling the corresponding combo with parameter of items separated by zeros, but it's not _bytes()-ed (char-ed) and it complains about str/bytes. The question is whether it's good to rework the combo function to use the combo with the char** parameter in such way as the listbox function or add just encode the items?

**Comment by swistakm** [comment]
I close this PR because the same changes were merged as part of #15

**Pull request closed without merging**